Core/automation/lib/python/core/date.py

```diff
@@ -34,87 +54,154 @@ def remove_java_converter(clazz):
             del clazz.__tojava__
     remove_java_converter(datetime.date)
     remove_java_converter(datetime.datetime)
-    
-from org.joda.time import DateTime, DateTimeZone
-from java.util import Calendar, Date, TimeZone
-from java.time import LocalDateTime, ZonedDateTime, ZoneId, ZoneOffset
+
+from java.time import LocalDateTime as javaLocalDateTime, ZonedDateTime as javaDateTime
+from java.time import ZoneId as javaZoneId, ZoneOffset as javaZoneOffset
 from java.time.format import DateTimeFormatter
 from java.time.temporal.ChronoUnit import DAYS, HOURS, MINUTES, SECONDS
-from org.openhab.core.library.types import DateTimeType as LegacyDateTimeType
-from org.eclipse.smarthome.core.library.types import DateTimeType
+from datetime import datetime as pythonDateTime, date as pythonDate
+from org.joda.time import DateTime as jodaDateTime, DateTimeZone as jodaDateTimeZone
+from java.util import Calendar as utilCalendar, Date as utilDateTime, TimeZone as utilTimeZone
+from org.eclipse.smarthome.core.library.types import DateTimeType as eclipseDateTime
+
+try:
+    # if the compat1.x bundle is not installed the 1.x DateTimeType is not available
+    from org.openhab.core.library.types import DateTimeType as legacyDateTime
+except:
+    pass
 
-__all__ = ["ZonedDateTime", "format_date", 
-           "days_between", "hours_between", "minutes_between", "seconds_between",
-           "to_java_zoneddatetime", "toJTime", "to_java_calendar", "toJCal", 
-           "to_python_datetime", "toPyDT", "pythonTimezone",
-           "to_joda_datetime", "toJodaDT"]
```

Comment:
Did you have name conflicts or did this just make it easier for you to understand the types? I don't like renaming these classes, especially if they are being made available through `__all__`.

Reply:
For clarity mostly since they all have almost the same name. Python and joda have the same name but Python is all lower case so there isn't a name conflict. I did it thinking about our target audience of non-programmers

Reply:
Users will see the function names. Developers will see the names you've made up for the classes when reading through the module.

Reply:
But we're exporting the classes for ease of use